### PR TITLE
[W-11031660] - Added processed Status logic

### PR DESCRIPTION
**Pull request by gerardorodriguezsf**
# Technical Design Review
[[Technical Design Review] W-11031660 - NPSP](https://salesforce.quip.com/iPNXA3ra3pZy)

# Changes
Update the RecurringDonationsService and RecurringDonationsSelector classes to process the recurring donation record data and build the processed Status data. Show this processed Status data in the Recurring Donations LWC.

# Work Closed
[W-11031660](https://gus.lightning.force.com/a07EE00000vTYLlYAO)

### CODE REVIEW

File: force-app/main/default/classes/RD2_ERecurringDonationsService.cls [hunk 1]
```diff
@@ -38,6 +38,9 @@ public with sharing class RD2_ERecurringDonationsService {
     private final String MONTHLY = 'Monthly';
     private final String WEEKLY = 'Weekly';
     private final String YEARLY = 'Yearly';
+    private final String STATUS_PAUSED = 'Paused';
```
Comment (lparrott) [inline force-app/main/default/classes/RD2_ERecurringDonationsService.cls:41]: These are included in `RD2_Constants` (`STATUS_PAUSED`, etc.)

Comment (gerardorodriguezsf) [reply force-app/main/default/classes/RD2_ERecurringDonationsService.cls:41]: OK Luke!, I'll add it. Thanks.

File: force-app/main/default/classes/RD2_ERecurringDonationsService.cls [hunk 2]
```diff
@@ -154,4 +158,30 @@ public with sharing class RD2_ERecurringDonationsService {
...
         return frequency;
     }
+
+    /** 
+    * @author Salesforce.org
+    * @date 2022
+    * @description process Recurring Donation record to build the status string.
+    * @return String returns the Status string.
+    */
+    private String getStatus(npe03__Recurring_Donation__c recurringDonation) {
```
Comment (lparrott) [inline force-app/main/default/classes/RD2_ERecurringDonationsService.cls:168]: I think you could replace this with the following:
`RD2_RecurringDonation rd = new RD2_RecurringDonation(recurringDonation);`
`return rd.getState();`

Comment (gerardorodriguezsf) [reply force-app/main/default/classes/RD2_ERecurringDonationsService.cls:163]: It's a processed State based on RD2_StatusMapper, Status__c field and the else case.

**Review by lparrott — APPROVED** [review]

**Pull request merged**